### DISCUSSION TITLE
Environment Variables not found after linking vercel/postgres db to project

### DISCUSSION THREAD

**Discussion by pixeldasher** [vercel/community#5912] [Help]
### Summary

I want to connect my vercel postgres db with my vercel project.

The environment variables were set when linking the db with my project, but for some reason the env vars can't be accessed/are missing in my production build.

I'm using vite for a QwikJS project and already checked whether or not my env vars are getting passed through and locally everything is working fine (.env, .env.local, .env.[mode].local, etc.) but here on vercel it is NOT working. I also passed down all the env vars via my vite config file with dotenv-expand (since this is needed in order to use env vars with vite). I triple checked everything and it works locally with no issues.

I tried searching for similar problems online and those questions were left unanswered or provided a fix that did not help me.

Again, all the env vars are present in the "Settings" tab but, alas, I get the same error code (VercelPostgresError: [...] no [...] env var was found) all the time.

### Steps to Reproduce

* Set up a vite (QwikJS) project
* Set up a vercel/postgres db
* Link db to project
* Use vercel/postgres npm package
* Build and deploy on vercel

**Comment by nicehatharry** [comment]
fwiw, I'm building a basic Vite project and using their solution of dotenv-expand in the vite.config to surface the DB env vars didn't work for me either.

**Comment by amyegan** [comment]
Do you get the same error for other environment variables added [in Project Settings](https://vercel.com/docs/projects/environment-variables#declare-an-environment-variable), or is this problem limited to Postgres variables?

**Comment by pixeldasher** [comment]
The error occures not only with the vercel/postgres variables but with all env vars, sadly.

I even tried setting the postgres variables by my own instead of letting them be generated, but that also did not help!

**Reply by amyegan** [reply]
Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share? I just built a starter Qwik app to test this and I couldn't replicate the issue. Both build-time variables and server-side variables are coming though as expected based on the [Qwik documentation](https://qwik.dev/docs/env-variables/). 

My example `onGet` function gets a build-time variable and a server-side variable and correctly returns the values I have in project settings: `{"buildVariable":"This is a build-time variable","serverVariable":"This is a server-side variable"}`

<img width="893" alt="image" src="https://github.com/vercel/community/assets/8247105/1965868d-d176-4e21-ab86-05041c630b79">


https://qwik.dev/docs/env-variables/

**Reply by nicehatharry** [reply]
I don't have a Qwik project, but my basic Vite project is having the same error.

function.ts
```
import { sql, QueryResultRow } from '@vercel/postgres'
export async function getUsers(): Promise<QueryResultRow> {
	const { rows } = await sql`
	SELECT * FROM users;
`
	return rows
}
```
<br>

Component.tsx
```

export const GetUsersComponent = () => {

	const handleClick = async () => {
		await getUsers()
			.then((users) => {
				if (users) {
					console.log(users)
				}
			})
			.catch((err) => {
				console.error(`ERROR: ${err}`)
			})
	}

	return (
		<>
					<Button onClick={handleClick}>
						Get Users
					</Button>
		</>
	)
}
```

As I'm reading the Vercel Functions functionality, the `getUsers` function should run server-side with Vercel and have access to `process` and the necessary environment variables, yes?

The environment variables are visibly set in the project settings, but I still get this error.

>  VercelPostgresError: VercelPostgresError - 'missing_connection_string': You did not supply a 'connectionString' and no 'POSTGRES_URL' env var was found.

**Reply by amyegan** [reply]
Yes, serverless functions would run as server-side code. Vite uses a special `import.meta.env` object. You may need to modify some of the code to match the Vite way of accessing environment variables as shown in their docs: https://vitejs.dev/guide/env-and-mode

**Reply by nicehatharry** [reply]
@amyegan, thanks for looking into this. The problem is actually with the vercel/postgres library trying to access process.env.POSTGRES_URL. That's not exactly my code to change. Vite's import.meta.env nonsense irritates me too. I was hoping that building my postgres code in a serverless function would give it access to the process.env dictionary and allow the vercel/postgres library to pick up the necessary settings vars. Is there a workaround?

fwiw: I do have the code mentioned here in my vite.config.ts (https://vercel.com/docs/storage/vercel-postgres/limits#troubleshooting-nuxt-sveltekit-and-vite)
